Issue/multiple patient entry details

nyashabryan opened the pull request:
When there were multiple patient entries for the same patient, the patient view would raise an error because it used the PatientEntry.objects.get() method which raises an error when there are multiple objects of the same details

nyashabryan committed:
Fixed improperly matched tags

nyashabryan committed:
Patient view can now take care of multiple entries

nyashabryan committed:
Tests for the new patient view

nathanbegbie commented on cspatients/tests.py:
```diff
@@ -312,6 +312,77 @@ def test_get_view(self):
+            template_name="cspatients/patient.html"
+        )
+        self.assertInHTML(
+            "Sorry. No patient such patient was found.",
```
Could we re-word this? I think it should be "Sorry. No such patient was found."

nyashabryan replied:
Yes it should.

nathanbegbie commented on cspatients/tests.py:
```diff
@@ -312,6 +312,77 @@ def test_get_view(self):
+            decision_time=timezone.now()
+        )
+
+        response = self.client.get("/cspatients/patient/XXXXXX")
```
I might be getting a bit nit picky here, but I think this code might be improved by doing the following:
`response = self.client.get("/cspatients/patient/{}".format(jane.patient_id))`

This helps to do 2 things:
- makes things slightly more DRY
- communicates a bit more clearly what the XXXXXX would represent

I'll leave this one up to you though.

nyashabryan replied:
@nathanbegbie I think I will follow the suggestion. Makes much more sense to be clear.

nathanbegbie commented on cspatients/views.py:
```diff
@@ -59,17 +59,19 @@ def view(request):
-            "patiententry": False,
-        }
-        status_code = 400
+    except PatientEntry.MultipleObjectsReturned:
```
I just want to check the situation we are handling here - if a patient is entered into the system twice (or comes to the hospital more than once?), it will return the most recently changed entry?

nyashabryan replied:
So suppose a patient gets an operation, but after that is booked for another one. Rowan explained a situation where a patient can get an operation done, but maybe because of a complication has to be booked for another. This means that for that same patient there will be another entry for the other operation. Now if there are multiple entries for the same patient, using get on the queries won't work. I am now considering maybe it is better to ways assume it returns multiple entries and pick the latest one?

nathanbegbie replied:
Yeah, so I think that instead of catching an exception, you could simply run something like the following:
`patiententry = PatientEntry.objects.filter(patient_id=patient_id).order_by('-decision_time').first()`

nathanbegbie reviewed (changes requested):
@nyashabryan have left some comments - some things I think you should change and the other is up to you.
Just a heads up - it's quite helpful to write some sort of brief description in a PR that describes what you are trying to achieve or fix and in some cases, exmplaining your approach to fixing the issue is also helpful.

nyashabryan committed:
Made patient url from test more explicit

nyashabryan committed:
Fixed typo

nyashabryan commented:
With the last push I made, I have had to check for if the PatientEntry object is None or set and I removed the except block because using filters to get objects does not return the PatientEntry.DoesNotExist exception. So for me to return the right status code, I added the conditional statement to check if PatientEntry is None or not.

nyashabryan committed:
Extract one patiententry out of many

nyashabryan committed:
Fixed statement layout
